src/Runner.Worker/action_yaml.json

```diff
@@ -252,6 +256,34 @@
                 "loose-key-type": "non-empty-string",
                 "loose-value-type": "string"
             }
+        },
+        "job-defaults": {
```

Comment:
can we have the idea of a `composite-defaults` keyword here instead of `job-defaults`

Reply:
We currently have two defaults

`workflow-defaults` which apply to the entire run, and `job-defaults` which apply to a job. we should have a new idea of `composite-defaults` which apply only to the current composite action (not its children or its parents).

So if you set a default shell any of your immediate run steps can use it, but it won't use the job defaults, workflow defaults or parent composite action defaults.